network/common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -163,8 +167,23 @@ public TransportClient createClient(String remoteHost, int remotePort) throws IO
         } else {
           logger.info("Found inactive connection to {}, creating a new one.", address);
         }
+      } else {
+        if ((clientPool.clientsFailTime[clientIndex] != 0L) && (clientCreateTime < clientPool.clientsFailTime[clientIndex])) {
```

Comment:
should we add a few seconds when comparing clientCreateTime versus latest FailedTime? For example - 
clientCreateTime < clientPool.clientsFailTime[clientIndex] + 5 * 1000 // whether the connection timed-out in last 5s